cmdLine/src/main/scala/topgun/cmdline/FileParser.scala

```diff
@@ -12,63 +12,95 @@ class FileParser(file: File, cmdLine: JfrParseCommandLine, totals: Totals, confi
 
   import configuration.{includeStack, includeThread}
 
-
+  var classPath:Option[String] = None
 
   def parse(): Unit = {
 
     println(s"*** File $file")
-    var allocationInNewTlab: Boolean = false
-    var allocationOutsideTlab: Boolean = false
-    var methodProfilingSample: Boolean = false
-    try {
+    var classPaths =  new util.ArrayList[URL]
+
+    def preProcessing(): Unit = {
+
+      var AllocationInNewTlabEnabled: Option[Boolean] = None
```

Comment:
if the range of values is not set or Some(true) we dont need the optionality

Reply:
the variable has three states,
None   => we have not seen the the recording setting for that event yet
true => we have seen the settings and it is true
false => we have seen the settings and it is false
It helps to know when it is done reading the events we want.